### ISSUE TITLE
redraw after remote changes

### ISSUE THREAD

**Issue by friedolino78** [mruby-zest/mruby-zest#16]
I observed, that the redraw mechanisms that work, if changing via gui, dont work if changing on the remote (via midi).

as a workaround for my special use case I added the whenValue call locally to the setValue Method the drive knob:
```
                Knob {
                    extern: dst.extern + "Distorsion/Pdrive"; label: "drive"
                    whenValue: lambda {wave.refresh};
                    function setValue(v) {
						valuator.value = lim(v, 0.0, 1.0);
						valuator.whenValue.call;
						valuator.damage_self
                    }
                }
```

I thought of an alternative solution:
calling the whenValue in valuator itself but checking whether the value actually changed to prevent loops.

**Comment by fundamental** [comment]
The alternative solution is not viable as there's possibly multiple 'values' that exist for a given parameter at a time. Consider moving a knob. It produces several values which are transmitted to the synthesis engine in order. Using the different-value heuristic fails if you capture value1 at time t1 and value 2 at time t2 as you want to avoid:

- send value 1
- send value 2
- observe value 1 as a response
- send value 1
- observe value 2 as a response
- send value 2
- repeat from step 3

**Comment by friedolino78** [comment]
I think the problem arises when signal propagation and signal feedback use the same methods.
Maybe this is the case for the whenValue method?

**Comment by fundamental** [comment]
Yep, there's a bit of a mismatch in what might be expected at first due to the async/asymmetric nature of propagating some values around.
